src/term.rs

```diff
@@ -574,6 +577,47 @@ impl Term {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct SharedTerm {
+    term: Rc<Term>,
```

Comment:
Nitpick: it's a bit confusing to have `RichTerm`  having a `term` field, which itself has a `term` field. Maybe we could rename them respectively `shared` and `inner`, or something else. That's just a remark though, that's not very important right now.

Reply:
changed to `shared`